File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    length = len(goal)
    for word in range(length):
        if cards1 and goal[0] == cards1[0]:
            goal.pop(0)
            cards1.pop(0)
        elif cards2 and goal[0] == cards2[0]:
            goal.pop(0)
            cards2.pop(0)
    return "Yes" if not goal else "No"

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "";
        int c1=0;
        int c2=0;
        int g=0;
        for (String word: goal){
            if(c1 < cards1.length && cards1[c1].equals(goal[g])){
                c1++;
                g++;
            }else if(c2 < cards2.length && cards2[c2].equals(goal[g])){
                c2++;
                g++;
            }else{
                
               return "No";
            }
        }
        return "Yes";
    }
}